### PR TITLE
Fix race for nomos log service

### PR DESCRIPTION
**Pull request by al8n**
When write test case for integration test, I found that the log service is not concurrent safe. Here is the example lead to panic.

```rust
fn run_mixnode(name: &'static str, listen_addr: SocketAddr, log_dir: &PathBuf) {
    let log_dir = log_dir.clone();
    Builder::new()
        .name(name.to_string())
        .spawn(move || {
            ...
            let app = OverwatchRunner::<MixNode>::run(
                MixNodeServiceSettings {
                    node: config,
                    logging: LoggerSettings {
                        backend: LoggerBackend::File {
                            directory: log_dir.clone(),
                            prefix: Some(name.into()),
                        },
                        format: LoggerFormat::Plain,
                        level: Level::INFO,
                    },
                },
                None,
            )?;
            ...
        })
        .unwrap();
}

run_mixnode("mixnode1", "127.0.0.1:7707".parse().unwrap(), &dir);
run_mixnode("mixnode2", "127.0.0.1:7717".parse().unwrap(), &dir);
```

=

### CODE REVIEW

File: nomos-services/log/src/lib.rs [hunk 1]
```diff
@@ -144,6 +144,10 @@ macro_rules! registry_init {
 #[async_trait::async_trait]
 impl ServiceCore for Logger {
     fn init(service_state: ServiceStateHandle<Self>) -> Result<Self, overwatch_rs::DynError> {
+        use std::sync::Once;
+
+        static ONCE_INIT: Once = Once::new();
```
Comment (danielSanchezQ) [inline nomos-services/log/src/lib.rs:149]: This static should probably go under the same `cfg(test)`, the import as well.

**Review by danielSanchezQ — APPROVED** [review]
I do not really like this, but I do not know of a different solution atm. Would you mind opening an issue to try to find a different solution?

Anyway, good catch, thanks!

**Comment by danielSanchezQ** [comment]
@al8n Check the lints and made sure everything is green before merging!!

**Comment by al8n** [comment]
> @al8n Check the lints and made sure everything is green before merging!!

```
Error:   --> nomos-core/src/vote/mock.rs:26:18
   |
26 | pub struct Error(String);
   |            ----- ^^^^^^
   |            |
   |            field in this struct
   |
   = note: `-D dead-code` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(dead_code)]`
help: consider changing the field to be of unit type to suppress this warning while preserving the field numbering, or remove the field
   |
26 | pub struct Error(());
```
The lint error is not related to this PR, should I fix the lint in this PR?

**Comment by al8n** [comment]
> I do not really like this, but I do not know of a different solution atm. Would you mind opening an issue to try to find a different solution?

Open it: https://github.com/logos-co/nomos-node/issues/619

**Comment by danielSanchezQ** [comment]
> fix the lint in this PR?

No, if it is not related is ok. You can try to fix in master though.

**Comment by al8n** [comment]
> > fix the lint in this PR?
> 
> No, if it is not related is ok. You can try to fix in master though.

Fix clippy in https://github.com/logos-co/nomos-node/pull/620

**Pull request merged**